FileField storage parameter should also accept a callable

# Bug report

<!--
Hi, thanks for submitting a bug. We appreciate that.

But, we will need some information about what's wrong to help you.
-->

## What's wrong

When running mypy against a code base with django-stubs installed I get an unexpected typing error:

```
.../models.py:30: error: Argument "storage" to "FileField" has incompatible type "Callable[[], Storage]"; expected "Optional[Storage]"  [arg-type]
```

## How is that should be

The `storage` parameter of the `FileField` class can take either an instance of `Storage` or a no-arg callable that will return an instance of `Storage` as stated here: https://docs.djangoproject.com/en/3.1/ref/models/fields/#django.db.models.FileField.storage

## System information

- OS:
- `python` version: 3.8.3
- `django` version: 3.1
- `mypy` version: 0.782
- `django-stubs` version: 1.5.0

--- a/django-stubs/db/models/fields/files.pyi
+++ b/django-stubs/db/models/fields/files.pyi
@@ -39,7 +39,7 @@ class FileField(Field):
     def __init__(
         self,
         upload_to: Union[str, Callable, Path] = ...,
-        storage: Optional[Storage] = ...,
+        storage: Optional[Union[Storage, Callable[[], Storage]]] = ...,
         verbose_name: Optional[Union[str, bytes]] = ...,
         name: Optional[str] = ...,
         max_length: Optional[int] = ...,

--- a/django-stubs/forms/forms.pyi
+++ b/django-stubs/forms/forms.pyi
@@ -70,7 +70,12 @@ class BaseForm:
     def visible_fields(self): ...
     def get_initial_for_field(self, field: Field, field_name: str) -> Any: ...
     def _html_output(
-        self, normal_row: str, error_row: str, row_ender: str, help_text_html: str, errors_on_separate_row: bool,
+        self,
+        normal_row: str,
+        error_row: str,
+        row_ender: str,
+        help_text_html: str,
+        errors_on_separate_row: bool,
     ) -> SafeText: ...
 
 class Form(BaseForm):

